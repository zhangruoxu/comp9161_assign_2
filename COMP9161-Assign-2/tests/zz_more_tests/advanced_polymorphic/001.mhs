main = 
  let g = letfun f x = 
            if (x > 1) 
              then (let h = letfun g y = x; in h) 
              else (let h = letfun g y = x; in h) ; 
  in g;
