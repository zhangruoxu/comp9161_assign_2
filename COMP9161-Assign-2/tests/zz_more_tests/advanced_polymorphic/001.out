main :: forall k. (Int -> k -> Int) = let g :: forall j. (Int -> j -> Int) = (letfun f :: (Int -> j -> Int) x = if (>) x 1
                                                                                                                   then let h :: forall e. (e -> Int) = (letfun g :: (e -> Int) y = x);
                                                                                                                        in h
                                                                                                                   else let h :: forall h. (h -> Int) = (letfun g :: (h -> Int) y = x);
                                                                                                                        in h);
                                      in g;
